specification.txt

```diff
@@ -1646,28 +1648,39 @@ are always tried in the listed order:
             affected, and disambiguation stops after the first name that
             eliminates cite ambiguity.
 
-``disambiguate-add-year-suffix`` [Step (3)]
+In the description of disambiguation methods (1) and (2) above, we assumed that
+each (disambiguated) cite has an unambiguous link to its bibliographic entry. To
+assure that each cite does in fact uniquely identify its entry in the
+bibliography, detail that distinguishes cites (such as names, initials, and full
+given names) must be shown in the corresponding bibliography entries. If this is
+not the case, disambiguation methods (1) and (2) also act on all members of a
+set of ambiguously cited bibliographic entries, until no more entries in the set
+can be unambiguously cited by adding (expanded) names. Each method only takes
+effect on the involved bibliographic entries after it has been used to
+disambiguate cites.
+
+A disambiguation attempt can also be made by rendering ambiguous cites with the
+``disambiguate`` condition testing "true" [Method (3)] (see `Choose`_).
```

Comment:
I don't understand what these lines mean. What are some examples here?

Reply:
Pinging @rmzelle @cormacrelf @adam3smith @fbennett. Could you folks take a look at thes?

Reply:
I'm not sold. There are two things going on. One is that if your cite has to add initials/given names to unambiguously refer to a bib entry, then the bib entry should also end up with initials/givens. The way this was ensured previously is that people would have bibliographies with full names in them. Are there even any styles that don't do this? What sort of bibliography declines to print authors' full names? Also it's a bit tricky to implement (but not impossible) -- is it a real problem that people are experiencing? Are there actual style guides out there that require cite-style name disambiguation within the bibliography itself in order to safely use initials? Because if that's not the case, don't bother reading the rest here, it's not worth your time :) But generally the interpretation I've managed to eke out of the rest seems like a lot of work before I've seen it to be a problem at all.

It's a bit of a slog to make sense of the rest, but I guess that's the game we're playing. The best reading I can do is that *"The overall goal is for bib entries to be uniquely citeable such that a person reading the document can tell which entry a cite refers to, so we try to expand or add names in bib entries until the ambiguous cites that refer to them stop getting any less ambiguous."* Honestly, you can stop there, you don't need to be more detailed than that in the disambiguation spec and I'd prefer we didn't. I think what's meant by "Each method only takes effect..." is that the un-expanded versions of the rendered cites (i.e. `<citation>`) built specifically for disambiguation against a specific entry should NOT stop being the basis for cites matching that entry during cite disambiguation, but you can just say "This does not affect the cite disambiguation process.". What this requires is:

1. add/expand a name in a bibliography entry
2. look up the same name in the dummy cites used to match other cites against, propagate the changes
3. run through all the cites that matched previously again to check if any of them would be more finely disambiguated after the change. You can ignore the ones that actually refer to the entry.
4. stop when they can't.

I think the intention is that this improves the yield for cite-disambiguation. You'd have to mount a tricky argument to show me it would help, because cite disambiguation is meant to stop just before it ceases to be effective; if you've already added the names/initials/givens from all the cites referred to in step (3) already (in their "corresponding bibliography entries"), can you even devise a test case that will trigger this uniqueness failsafe? I have no energy left to do that.

Altogether, it's like... OK... but disambiguation is already slow. This not only adds a pretty big computation, it makes it more difficult to make the existing stuff fast. If you think of matching a cite against a possible matching reference as a list of normalised, rendered cites built with a dummy cite for that reference, then you are no longer so free to optimise that list. (With your weapon of choice, DFAs, RegexSet, ...) You have to have the ability to do step (2) on those dummy rendered cites, and none of those code weapons can withstand that mutation. That list was one of the very few things that never changes as long as a reference doesn't change! So -- render them again, and each time you add a single initial to the bib entry, render more? At the very least these things are not "the same disambiguation methods" as for cites. They'll have to be re-written to reflect the is-cite-ambiguity-improved test. Any way you cut it, this is a LOT of work that's already solved by using full names in your bibliographies.